qiskit/visualization/circuit/_utils.py

```diff
@@ -611,17 +613,17 @@ def slide_from_left(self, node, index):
             index_stop = -1
             if (condition := getattr(node.op, "condition", None)) is not None:
                 index_stop = max(
-                    (self.measure_map[bit] for bit in condition_resources(condition).clbits),
+                    (
+                        self.measure_map[bit]
+                        for bit in condition_resources(condition).clbits
+                        if bit in self.measure_map
+                    ),
```

Comment:
I am now convinced that for your fix to work, we don't need to change any function except for `_get_gate_span`. So I suggest to return this to the original version. (I do think that there is quite a bit of opportunity to optimize and improve this visualization code, but that shouldn't be a part of this PR).